core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -104,27 +104,30 @@ object ZkUtils {
    * Get calls that only depend on static paths
    */
   def getTopicPath(topic: String): String = {
-    ZkUtils.BrokerTopicsPath + "/" + topic
+    getZkPath(ZkUtils.BrokerTopicsPath, topic)
   }
 
   def getTopicPartitionsPath(topic: String): String = {
-    getTopicPath(topic) + "/partitions"
+    getZkPath(getTopicPath(topic), "partitions")
   }
 
   def getTopicPartitionPath(topic: String, partitionId: Int): String =
-    getTopicPartitionsPath(topic) + "/" + partitionId
+    getZkPath(getTopicPartitionsPath(topic), partitionId)
 
   def getTopicPartitionLeaderAndIsrPath(topic: String, partitionId: Int): String =
-    getTopicPartitionPath(topic, partitionId) + "/" + "state"
+    getZkPath(getTopicPartitionPath(topic, partitionId), "state")
     
   def getEntityConfigRootPath(entityType: String): String =
-    ZkUtils.EntityConfigPath + "/" + entityType
+    getZkPath(ZkUtils.EntityConfigPath, entityType)
 
   def getEntityConfigPath(entityType: String, entity: String): String =
-    getEntityConfigRootPath(entityType) + "/" + entity
+    getZkPath(getEntityConfigRootPath(entityType), entity)
 
   def getDeleteTopicPath(topic: String): String =
-    DeleteTopicsPath + "/" + topic
+    getZkPath(DeleteTopicsPath, topic)
+
+  def getZkPath(nodes: Any*): String =
```

Comment:
Thank you @ijuma for your feedback. To address your comments,

- I used `Any*` as the type of `nodes` because in a few instances non-string types are being used to build the path. Examples are [this](https://github.com/apache/kafka/pull/1025/files#diff-f1b330624af520e6f9824bd308e21f24L219) (an `Int` value) and [this](https://github.com/apache/kafka/pull/1025/files#diff-10f2e53763ecbd493fc649562dda6770L259) (a `ResourceType` value).

- I am trying to understand how your suggestion regarding how we should expose those utility methods. If we are to remove path building logic from callers I think it would mean the callers would have to pass on the nodes along with any method that would use those nodes and have the called method take care of building the path. For example,
`zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)`
would change to
`zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir, topicPartition.partition)`
where the actual path is built inside `readDataMaybeNull`. This would considerably expand the scope of the patch as quite a few methods like this would need to be (I hope) slightly modified.

I could be misunderstanding what you suggested, in which case, I appreciate your clarification.